### PR TITLE
Fix staticcheck linting problems

### PR DESCRIPTION
**Pull request by albertony**
#### What is the purpose of this change?

Fix more "problems" reported by the staticcheck lint tool (v0.3.2).

#### Was the change discussed in an issue or in the forum before?

Continuing from https://github.com/rclone/rclone/pull/6228
See also https://github.com/rclone/rclone/issues/6273

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [X] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [X] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/s3/s3.go [hunk 1]
```diff
@@ -3837,7 +3839,8 @@ func (o *Object) downloadFromURL(ctx context.Context, bucketPath string, options
 	for key, value := range resp.Header {
 		if strings.HasPrefix(key, "x-amz-meta") {
 			metaKey := strings.TrimPrefix(key, "x-amz-meta-")
-			metaData[strings.Title(metaKey)] = &value[0]
+			metaKey = cases.Title(language.Und, cases.NoLower).String(metaKey)
```
Comment (ncw) [inline backend/s3/s3.go:3842]: Gah, this must be wrong! Your change is fine, but the code is wrong before. I suggest you leave this one, and I will fix it properly in my metadata branch!

File: cmd/help.go [hunk 2]
```diff
@@ -315,7 +317,8 @@ func showBackend(name string) {
 			optionsType = "advanced"
 			continue
 		}
-		fmt.Printf("### %s options\n\n", strings.Title(optionsType))
+		optionsType = cases.Title(language.Und, cases.NoLower).String(optionsType)
```
Comment (ncw) [inline cmd/help.go:320]: This is really ugly! the strings are always ASCII strings for which `strings.Title` works properly.

File: fs/operations/operations_test.go [hunk 3]
```diff
@@ -270,7 +272,7 @@ func TestHashSums(t *testing.T) {
 		if !hashes.Contains(test.ht) {
 			continue
 		}
-		name := strings.Title(test.ht.String())
+		name := cases.Title(language.Und, cases.NoLower).String(test.ht.String())
```
Comment (ncw) [inline fs/operations/operations_test.go:275]: Likewise these are always ASCII strings...

File: backend/yandex/yandex.go [hunk 4]
```diff
@@ -782,11 +782,11 @@ func (f *Fs) DirMove(ctx context.Context, src fs.Fs, srcRemote, dstRemote string
...
 
 	_, err = f.readMetaDataForPath(ctx, dstPath, &api.ResourceInfoRequestOptions{})
-	if apiErr, ok := err.(*api.ErrorResponse); ok {
+	if _, ok := err.(*api.ErrorResponse); ok {
 		// does not exist
-		if apiErr.ErrorName == "DiskNotFoundError" {
-			// OK
-		}
+		//if apiErr.ErrorName == "DiskNotFoundError" {
+		//	// OK
```
Comment (ncw) [inline backend/yandex/yandex.go:788]: This should probably return the error if `apiErr.ErrorName != "DiskNotFoundError"`...

File: cmd/selfupdate/verify.go [hunk 5]
```diff
@@ -10,9 +10,9 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/ProtonMail/go-crypto/openpgp"
```
Comment (ncw) [inline cmd/selfupdate/verify.go:13]: How did you choose this package?

**Review by ncw — COMMENTED** [review]
I've been though these. See comments inline.

Thank you :-)

File: cmd/help.go (hunk 2 above)
Comment (albertony) [reply cmd/help.go:320]: It's not "neat", I can agree with that. Still, ascii or not, `strings.Title` is listed as deprecated on https://pkg.go.dev/strings#Title, so I would think it is better not to use it.

File: cmd/selfupdate/verify.go (hunk 5 above)
Comment (albertony) [reply cmd/selfupdate/verify.go:13]: There wasn't a whole lot of good alternatives to pick from. This was one which had recent commits, and came from a somewhat known company, with privacy and encryption related focus, though I cannot vouch for them any more then others..

The deprecation notice in https://pkg.go.dev/golang.org/x/crypto/openpgp and the linked issue there tries to steer users away from OpenPGP in general. The ProtonMail variant is mentioned at the of the issue description:

> Every major player that actually relies on OpenPGP and Go maintains a fork (https://github.com/keybase/go-crypto, https://github.com/ProtonMail/go-crypto, ...). Developers would be better served by finding those, and by centralizing efforts onto one, without blocking on the resource limits of the Go team...

If this is something that needs more consideration, maybe I should split it out in a separate PR? I'll probably be away a lot the coming weeks, so then I could try to get the rest merged..

File: cmd/help.go (hunk 2 above)
Comment (ncw) [reply cmd/help.go:320]: OK. Let's go with the flow!

**Comment by albertony** [comment]
Updated according to comments. I assume this can be merged now...?

**Review by ncw — APPROVED** [review]
LGTM - please merge

**Pull request merged**